### PR TITLE
Shugeo random uniform int

### PR DESCRIPTION
**Pull request by shugeo**
## What changes were proposed in this pull request?

Added integer part for uniform distribution op.

## How was this patch tested?

Proper test was added.

## Quick checklist

The following checklist helps ensure your PR is complete:

- [ ] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [ ] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [ ] Created tests for any significant new code additions.
- [ ] Relevant tests for your changes are passing.

### CODE REVIEW

File: libnd4j/include/graph/RandomGenerator.h [hunk 1]
```diff
@@ -194,8 +194,22 @@ namespace nd4j {
...
+        template <>
+        _CUDA_HD FORCEINLINE Nd4jLong RandomGenerator::relativeT(Nd4jLong index, Nd4jLong from, Nd4jLong to) {
+            auto t = this->relativeT<double>(index);
+            auto z = from + Nd4jLong(t * (to - from));
+            return z;
+        }
+
+        template <>
+        _CUDA_HD FORCEINLINE int RandomGenerator::relativeT(Nd4jLong index, int from, int to) {
+            auto t = this->relativeT<float>(index);
```
Comment (raver119) [inline libnd4j/include/graph/RandomGenerator.h:211]: Why not something like this:
```
auto t = this->relativeT<int>(index);
```
?

File: libnd4j/include/ops/declarable/generic/random/uniform.cpp [hunk 2]
```diff
@@ -35,41 +36,40 @@ namespace nd4j {
          * TArgs[0] - min for rng
          * TArgs[1] - max for rng
          */
-        CUSTOM_OP_IMPL(randomuniform, 1, 1, true, 2, 0) {
+        CUSTOM_OP_IMPL(randomuniform, 3, 1, true, 0, 0) {
```
Comment (raver119) [inline libnd4j/include/ops/declarable/generic/random/uniform.cpp]: No. revert it back here. min/max arrays must be optional, as well as INT_ARGS/T_ARGS

**Review by raver119 — CHANGES_REQUESTED** [review]

File: libnd4j/include/graph/RandomGenerator.h (hunk 1 above)
Comment (shugeo) [reply libnd4j/include/graph/RandomGenerator.h:211]: float value from 0 to 1 multiply to range and convert to int. As result - int value distributed between from and to uniformly.

File: libnd4j/include/ops/declarable/helpers/cuda/random.cu [hunk 3]
```diff
@@ -181,6 +181,56 @@ namespace helpers {
...
+            RandomLauncher::fillUniform(context, rng, output, minVal, maxVal);
+        else {
+            auto stream = context->getCudaStream();
+            graph::RandomGenerator *devRng;
+            cudaMalloc(&devRng, sizeof(graph::RandomGenerator));
+            cudaMemcpy(devRng, &rng, sizeof(graph::RandomGenerator), cudaMemcpyHostToDevice);
+            auto outputBuf = output->dataBuffer()->specialAsT<T>();
+            auto outputShape = output->specialShapeInfo();
+            fillUniformKernel<T><<<128, 128, 128, *stream>>>(devRng, minVal, maxVal, outputBuf, outputShape);
+            cudaFree(devRng);
```
Comment (raver119) [inline libnd4j/include/ops/declarable/helpers/cuda/random.cu]: synchronize before cudaFree call

**Review by raver119 — CHANGES_REQUESTED** [review]

File: libnd4j/include/ops/declarable/generic/random/uniform.cpp (hunk 2 above)
Comment (shugeo) [reply libnd4j/include/ops/declarable/generic/random/uniform.cpp]: Yes, changed to use min/max vals as optional with default min = 0, max = +inf

File: libnd4j/include/ops/declarable/helpers/cuda/random.cu (hunk 3 above)
Comment (shugeo) [reply libnd4j/include/ops/declarable/helpers/cuda/random.cu]: Added error checking and synchronize stuff.

**Review by raver119 — APPROVED** [review]

**Pull request merged**